### ISSUE TITLE
[FEAT]: Require approval from someone other than the last pusher

### ISSUE THREAD

**Issue by tatutoivio-bf4d** [integrations/terraform-provider-github#1352]
### Describe the need

I see that at least according to the `branch_protection` [documentation](https://registry.terraform.io/providers/integrations/github/5.7.0/docs/resources/branch_protection) of the currently latest version (5.7.0) the "Require approval from someone other than the last pusher" option isn't available for the `required_pull_request_reviews` parameter.

That boolean option would be very important for my use case, and its name could be e.g. `require_non_last_pusher_approval`.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tatutoivio-bf4d** [comment]
Looking at the GitHub API, I can see that the proper name for the option might rather be `require_last_push_approval`.

**Comment by fproulx-boostsecurity** [comment]
This is such an important improvement to branch protection. I hope this will go in soon.

**Comment by kfcampbell** [comment]
I agree, this would be awesome to have! Do either of you have interest in opening a PR for this behavior?

**Comment by wwsean08** [comment]
Looking into this, it looks like `github.com/shurcooL/githubv4` will have to be updated first to support this new property unless I'm misinterpreting the code.  I'm going to try and work on getting a PR against that repo this weekend to add the field.  There are actually a couple of new fields missing which I'm going to try to get into the `githubv4` library for branch protections.  I'll post the PR link here if I'm able to get one opened over there since it's a pre-requisite for this work.

**Comment by wwsean08** [comment]
Actually no need for me to open a PR, this one takes care of the necessary change, https://github.com/shurcooL/githubv4/pull/107

**Comment by wwsean08** [comment]
I've got a [commit here](https://github.com/integrations/terraform-provider-github/commit/fe87659713f1177031b761f84d2dc716c3641761) that does all the work to add this option as well as the lock branch option since they were introduced at the same time.  Because my test requires overriding the current version of the `githubv4` library I haven't bothered opening a PR, but this should give someone a good start if someone else wants to take it up once this is merged.  I imagine you will be able to get away with leaving the go.mod at 1.16, but I had trouble which I think had to do with the go.mod (specifically `golang.org/x/oauth2 v0.1.0`) in the `githubv4` repo which seemed to need a package introduced in 1.17, but I didn't dig too deep into the issue, so your results may vary.

If I'm able to open a PR once that (or another PR on githubv4 gets merged) I will, but I am not able to spend as much time on this as I would like so hopefully my code snippet above helps someone down the line.  Below is the code I wrote to test and validate it so it _should_ work along with a screenshot of the results:

```
terraform {
  required_providers {
    github = {
      source  = "integrations/github"
      version = "~> 5.0"
    }
  }
}

provider "github" {
  owner = "REDACTED"
}

data "github_repository" "repo" {
  full_name = "REDACTED/REDACTED"
}

resource "github_branch_protection" "protection" {
  repository_id              = data.github_repository.repo.node_id
  pattern                    = "test"
  lock_branch                = true
  required_pull_request_reviews {
    require_last_push_approval = true
  }
}
```

![Screen Shot 2022-11-16 at 3 36 50 PM](https://user-images.githubusercontent.com/839261/202319726-85ffa762-002f-43ba-878b-4c5517999d16.png)

**Comment by wwsean08** [comment]
Good news, today a [PR](https://github.com/shurcooL/githubv4/pull/110) I opened that just updates the schema got merged 🎉 .  Friday I will look to open a simplified version of my previous change as a PR to hopefully get this feature implemented
